packages/blocks/src/types/block.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Block Interface
+ *
+ * @module blocks/Block
+ */
+
+/**
+ * Internal dependencies
+ */
+import type { BlockNode } from './block-node';
+
+/**
+ * Instance of a [BlockType](./block-type.ts).
+ *
+ * @public
+ */
+export interface Block< Attributes extends Record< string, any > = {} > {
+	/**
+	 * The block's registered name.
+	 */
+	name: string;
+
+	/**
+	 * Attributes for the block.
+	 */
+	attributes: Attributes;
```

Comment:
```ts
/**
 * Description of block's attribute types.
 *
 * These define how to source the attributes for a block and
 * may provide default values in case they aren't available.
 *
 * Example
 *     {
 *       "url": {
 *         "type": "string",
 *         "source": "attribute",
 *         "selector": "img",
 *         "attribute": "src"
 *       }
 *     }
 * 
 * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/
 */
```